src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -24,406 +33,186 @@ class Kalman(BayesFiltSmooth):
         If an DiscreteLTIGaussian, the measurement matrix is :math:`L \\times K` dimensional, the forcevec is `L` dimensional and the meascov is `L \\times L` dimensional.
     initrv
         Initial random variable for the prior. This is a `K` dimensional Gaussian distribution (not `L`, because it belongs to the prior)
+    predict
+        Custom prediction step. Choose between e.g. classical and square-root implementation. Default is 'predict_via_transition`.
+    measure
+        Custom measurement step. Choose between e.g. classical and square-root implementation. Default is 'measure_via_transition`.
+    update
+        Custom update step. Choose between e.g. classical, Joseph form, and square-root implementation. Default is 'update_classic`.
+    smooth_step
+        Custom update step. Choose between e.g. classical, and square-root implementation. Default is 'rts_smooth_step_classic`.
     """
 
-    def __init__(self, dynamics_model, measurement_model, initrv):
-        """Check that the initial distribution is Gaussian."""
-        if not issubclass(type(initrv), Normal):
-            raise ValueError(
-                "Gaussian filters/smoothers need initial "
-                "random variables with Normal distribution."
-            )
-        super().__init__(dynamics_model, measurement_model, initrv)
-
-    def filtsmooth(self, dataset, times, _intermediate_step=None):
-        """Apply Gaussian filtering and smoothing to a data set.
+    def __init__(
+        self,
+        dynamics_model,
+        measurement_model,
+        initrv,
+        predict=predict_via_transition,
+        measure=measure_via_transition,
+        update=update_classic,
+        smooth_step=rts_smooth_step_classic,
+    ):
+        self.dynamics_model = dynamics_model
+        self.measurement_model = measurement_model
+        self.initrv = initrv
+        self.predict = lambda *args, **kwargs: predict(dynamics_model, *args, **kwargs)
+        self.measure = lambda *args, **kwargs: measure(
+            measurement_model, *args, **kwargs
+        )
+        self.update = lambda *args, **kwargs: update(measurement_model, *args, **kwargs)
```

Comment:
nail down the inputs that never change.

Reply:
I see your point. I think it is a valid concern. I think the way to go would be to rename the arguments, not the attributes.
An underscore in the arguments might suggest that we dont want people to change this though, which I think is not the way to go.

How about something like `use_predict=...`, `use_measure=`, ...; then, `self.predict` could remain as is, and it would be clear that this is more of a rule that is passed on that the actual function. (Nb: I think `predict_rule`, `measure_rule`, etc., are not very good names). 

I will rename with `use_*`, and if you agree with this choice then please feel free to resolve. If not, please complain, so we can find a good solution!

Reply:
Sounds good :)